### PR TITLE
fixes #2324 getRequestURL(path) should not include current request's query string

### LINKED ISSUE #2324: getRequestURL(path) adds irrelevant queryString from current request
This line probably should be removed:
https://github.com/jooby-project/jooby/blob/34eb70cf2583e317d3c77f94d165bc8be7520bb5/jooby/src/main/java/io/jooby/DefaultContext.java#L252

If getRequestURL() is used with a different path than current request, then adding query string from current request is very unexpected and results in bugs, e.g. when implementing an OAuth2 client. Or this line could be moved to the empty path if branch, e.g. if user wants to get the current request's full URL.

### PR DESCRIPTION
**Pull request by cbxp**

### CODE REVIEW

File: jooby/src/main/java/io/jooby/DefaultContext.java [hunk 1]
```diff
@@ -228,10 +228,10 @@
   }
 
   @Override default @Nonnull String getRequestURL() {
-    return getRequestURL("");
+    return getRequestURL(null);
   }
 
-  @Override default @Nonnull String getRequestURL(@Nonnull String path) {
```
Comment (jknack) [inline jooby/src/main/java/io/jooby/DefaultContext.java:234]: let's keep the `@Nonnull` annotation. This always append the provided path (never the `getRequestPath` and never append the queryString

File: jooby/src/main/java/io/jooby/DefaultContext.java [hunk 2]
```diff
@@ -228,10 +228,10 @@
   }
 
   @Override default @Nonnull String getRequestURL() {
-    return getRequestURL("");
+    return getRequestURL(null);
```
Comment (jknack) [inline jooby/src/main/java/io/jooby/DefaultContext.java:231]: This one add call `getRequestURL(getRequestPath())` and append queryString if present

**Comment by jknack** [comment]
Looks good, couple of comment. Let's make sure javadoc is updated too

File: jooby/src/main/java/io/jooby/DefaultContext.java (hunk 1 above)
Comment (cbxp) [reply jooby/src/main/java/io/jooby/DefaultContext.java:234]: The annotation is still in the interface (for external consumers), but locally, there was a null check already, so why not pass a null as "undefined" value? Empty string can actually have a meaning of empty path

File: jooby/src/main/java/io/jooby/DefaultContext.java (hunk 2 above)
Comment (cbxp) [reply jooby/src/main/java/io/jooby/DefaultContext.java:231]: OK, this is probably a better idea, yes

**Comment by angryziber** [comment]
@jknack so, is it ok now to merge?

**Pull request merged**